run/run.go

```diff
@@ -175,7 +179,28 @@ func newHelp(e []string, c *cobra.Command) func(command *cobra.Command, strings
 	}
 }
 
-var version string
+func newFactory(cmd *cobra.Command) util.Factory {
+	flags := cmd.Flags()
+	kubeConfigFlags := genericclioptions.NewConfigFlags(true).WithDeprecatedPasswordFlag()
+	kubeConfigFlags.AddFlags(flags)
+	userAgentKubeConfigFlags := &cfgflags.UserAgentKubeConfigFlags{
+		Delegate:  kubeConfigFlags,
+		UserAgent: getUserAgent(),
+	}
+	matchVersionKubeConfigFlags := util.NewMatchVersionFlags(userAgentKubeConfigFlags)
+	matchVersionKubeConfigFlags.AddFlags(cmd.PersistentFlags())
+	cmd.PersistentFlags().AddGoFlagSet(flag.CommandLine)
+	return util.NewFactory(matchVersionKubeConfigFlags)
+}
+
+func getUserAgent() string {
+	if version == "" {
+		return "kpt/<unknown>"
+	}
+	return fmt.Sprintf("kpt/%s", version)
+}
+
+var version string = "v3.2.1"
```

Comment:
why are we setting this version?

Reply:
wdyt of just making the default version "uknown", and removing the ifs

Reply:
The version set here is just a leftover from some testing that I forgot to remove. But I like the suggestion of just setting it to unknown and remove the ifs.

Reply:
Fixed